_studio/shared/umc/codec/av1_dec/include/umc_av1_utils.h

```diff
@@ -0,0 +1,175 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+#include "umc_defs.h"
+#ifdef MFX_ENABLE_AV1_VIDEO_DECODE
+
+#ifndef __UMC_AV1_UTILS_H_
+#define __UMC_AV1_UTILS_H_
+
+#include "umc_av1_dec_defs.h"
+#include "umc_av1_frame.h"
+
+namespace UMC_AV1_DECODER
+{
+    inline uint32_t CeilLog2(uint32_t x) { uint32_t l = 0; while (x > (1U << l)) l++; return l; }
+
+    // we stop using UMC_VP9_DECODER namespace starting from Rev 0.25.2
+    // because after switch to AV1-specific segmentation stuff there are only few definitions we need to re-use from VP9
+    void SetSegData(SegmentationParams & seg, uint8_t segmentId, SEG_LVL_FEATURES featureId, int32_t seg_data);
+
+    inline int32_t GetSegData(SegmentationParams const& seg, uint8_t segmentId, SEG_LVL_FEATURES featureId)
+    {
+        return
+            seg.FeatureData[segmentId][featureId];
+    }
+
+    inline bool IsSegFeatureActive(SegmentationParams const& seg, uint8_t segmentId, SEG_LVL_FEATURES featureId)
+    {
+        return
+            seg.segmentation_enabled && (seg.FeatureMask[segmentId] & (1 << featureId));
+    }
+
+    inline void EnableSegFeature(SegmentationParams & seg, uint8_t segmentId, SEG_LVL_FEATURES featureId)
+    {
+        seg.FeatureMask[segmentId] |= 1 << featureId;
+    }
+
+    inline uint8_t IsSegFeatureSigned(SEG_LVL_FEATURES featureId)
+    {
+        return
+            SEG_FEATURE_DATA_SIGNED[featureId];
+    }
+
+    inline void ClearAllSegFeatures(SegmentationParams & seg)
+    {
+        std::fill_n(reinterpret_cast<int32_t*>(seg.FeatureData),
+            std::extent<decltype(seg.FeatureData), 0>::value * std::extent<decltype(seg.FeatureData), 1>::value, 0);
+        std::fill_n(seg.FeatureMask, std::extent<decltype(seg.FeatureMask)>::value, 0);
+    }
+
+    void SetupPastIndependence(FrameHeader & info);
+
+    inline bool FrameIsIntraOnly(FrameHeader const& fh)
+    {
+        return fh.frame_type == INTRA_ONLY_FRAME;
+    }
+
+    inline bool FrameIsIntra(FrameHeader const& fh)
+    {
+        return (fh.frame_type == KEY_FRAME || FrameIsIntraOnly(fh));
+    }
+
+    inline bool FrameIsResilient(FrameHeader const& fh)
+    {
+        return FrameIsIntra(fh) || fh.error_resilient_mode;
+    }
+
+    inline uint32_t NumTiles(TileInfo const & info)
+    {
+        return info.TileCols * info.TileRows;
+    }
+
+    int IsCodedLossless(FrameHeader const&);
+
+#if UMC_AV1_DECODER_REV >= 8500
+    inline void av1_load_previous(FrameHeader& fh, DPBType const& frameDpb)
```

Comment:
style fix: the most functions have pascal notation but this - snake one . Which style we use? Is there logic to use different ones?

Reply:
I will change it to pascal notation.